storage.go

```diff
@@ -135,7 +136,15 @@ func (s *Storage) fetch(ctx context.Context, path string, url string, opt pairSt
 	defer resp.Body.Close()
 
 	if resp.StatusCode != http.StatusOK {
-		return fmt.Errorf("fetch from url %s expected %d, but got %d", url, http.StatusOK, resp.StatusCode)
+		switch resp.StatusCode {
```

Comment:
We don't know the real meaning of those HTTP status codes, so don't need to handle them, just return ErrUnexpected for all of them.